src/coreComponents/physicsSolvers/fluidFlow/StencilDataCollection.cpp

```diff
@@ -0,0 +1,273 @@
+ */
+
+#include "StencilDataCollection.hpp"
+
+#include "common/Units.hpp"
+#include "finiteVolume/FluxApproximationBase.hpp"
+#include "constitutive/permeability/PermeabilityBase.hpp"
+#include "constitutive/permeability/PermeabilityFields.hpp"
+#include "physicsSolvers/fluidFlow/StencilAccessors.hpp"
+#include "mainInterface/ProblemManager.hpp"
+#include "physicsSolvers/PhysicsSolverManager.hpp"
+
+namespace geos
+{
+
+using namespace constitutive;
+using namespace dataRepository;
+
+
+StencilDataCollection::StencilDataCollection( const string & name,
+                                              Group * const parent ):
+  Base( name, parent )
+{
+  enableLogLevelInput();
+  getWrapperBase( Group::viewKeyStruct::logLevelString() ).
+    setDescription( "When higher than 1: Display store events details." );
+
+  registerWrapper( viewKeyStruct::solverNameString(), &m_solverName ).
+    setRTTypeName( rtTypes::CustomTypes::groupNameRef ).
+    setInputFlag( dataRepository::InputFlags::REQUIRED ).
+    setDescription( "Name of the flow solver" );
+
+  // registerWrapper( viewKeyStruct::connectionDataString(), &m_currentConnData );
+  registerWrapper( viewKeyStruct::cellAGlobalIdString(), &m_cellAGlobalId );
+  registerWrapper( viewKeyStruct::cellBGlobalIdString(), &m_cellBGlobalId );
+  registerWrapper( viewKeyStruct::transmissibilityABString(), &m_transmissibilityAB );
+  registerWrapper( viewKeyStruct::transmissibilityBAString(), &m_transmissibilityBA );
+}
+
+void StencilDataCollection::initializePostInitialConditionsPostSubGroups()
+{
+  ProblemManager & problemManager = this->getGroupByPath< ProblemManager >( "/Problem" );
+
+  { // find targeted solver
+    PhysicsSolverManager & physicsSolverManager = problemManager.getPhysicsSolverManager();
+
+    m_solver = physicsSolverManager.getGroupPointer< FlowSolverBase >( m_solverName );
+    GEOS_THROW_IF( m_solver == nullptr,
+                   GEOS_FMT( "{}: Could not find flow solver named '{}'.",
+                             getDataContext(),
+                             m_solverName ),
+                   InputError );
+  }
+
+  { // pre-allocate buffers
+    bool foundStencil = false;
+    DomainPartition & domain = problemManager.getDomainPartition();
+
+    m_solver->forDiscretizationOnMeshTargets( domain.getMeshBodies(),
+                                              [&] ( string const &,
+                                                    MeshLevel & mesh,
+                                                    arrayView1d< string const > const & )
+    {
+      NumericalMethodsManager const & numericalMethodManager = domain.getNumericalMethodManager();
+      FiniteVolumeManager const & fvManager = numericalMethodManager.getFiniteVolumeManager();
+      FluxApproximationBase const & fluxApprox = fvManager.getFluxApproximation( m_solver->getDiscretizationName() );
+
+      fluxApprox.forStencils< CellElementStencilTPFA >( mesh, [&]( auto const & stencil )
```

Comment:
Maybe supporting the output for all the other stencils could be useful too. I m thinking of debugging of fractured problems.